src/Builders/PromptBuilder.php

```diff
@@ -59,6 +59,11 @@ class PromptBuilder
      */
     protected ?ModelInterface $model = null;
 
+    /**
+     * @var list<ModelInterface|string> Preferred models to evaluate in order.
+     */
+    protected array $modelPreferences = [];
```

Comment:
Can we use the singular name here too (`modelPreference`)? It's more in line with the method, and using the plural here re-raises my previous concern from the issue about that name - `modelPreferences` could easily be confused with `modelOptions` or `modelConfig` or `modelSettings` as a concept.